### PR TITLE
feat: argmax + argmin

### PR DESCRIPTION
**Pull request by nathanielsimard**
+ Fix ndarray

### CODE REVIEW

File: burn-ndarray/src/ops/base.rs [hunk 1]
```diff
@@ -380,31 +380,46 @@ where
...
-    dim: usize,
     cmp: F,
 ) -> NdArrayTensor<i64, D>
 where
     F: Fn(&f64, &f64) -> Ordering,
 {
     let mut shape = tensor.shape();
-    let batch_size = shape.dims[dim];
-    let mut end = shape.dims[dim];
+    let batch_size = shape.dims[D - 1];
```
Comment (louisfd) [inline burn-ndarray/src/ops/base.rs:421]: I understand that the previous code was not accurate for other dims than D-1?

File: burn-wgpu/src/ops/numeric.rs [hunk 2]
```diff
@@ -170,20 +168,33 @@ impl<G: GraphicsApi> NumericOps<G> {
...
+        reduction_dim::<MeanDim, E, D>(tensor, dim)
+    }
+
+    pub fn argmax<E: WgpuElement + Element, I: WgpuElement, const D: usize>(
+        tensor: WgpuTensor<E, D>,
+        dim: usize,
+    ) -> WgpuTensor<I, D> {
+        reduction_args_dim::<ArgsMax, E, I, D>(tensor, dim)
+    }
+    pub fn argmin<E: WgpuElement + Element, I: WgpuElement, const D: usize>(
```
Comment (louisfd) [inline burn-wgpu/src/ops/numeric.rs:194]: No space between those functions lol. Isn't it auto-formatted?

**Review by louisfd — APPROVED** [review]
Good to go

File: burn-wgpu/src/ops/numeric.rs (hunk 2 above)
Comment (antimora) [reply burn-wgpu/src/ops/numeric.rs:194]: Yeah I think he missed it. Also currently WGPU is not hooked up to the CI.

**Review by antimora — CHANGES_REQUESTED** [review]
Looks good. Can you please run the fmt and clippy? It seems there is one formatting issue @louisfd identified.

File: burn-wgpu/src/ops/numeric.rs (hunk 2 above)
Comment (nathanielsimard) [reply burn-wgpu/src/ops/numeric.rs:194]: Nop, it's not a clippy not a fmt error/warning, eventhough I like to add spacing between functions.

**Pull request merged**